packages/odyssey/src/scss/base/_iconography.scss

```diff
@@ -0,0 +1,13 @@
+.ods-icon {
+  position: relative;
+  top: -0.1em;
+  width: 1em;
+  height: 1em;
+  vertical-align: middle;
+}
+
+.ods-button {
+  .ods-icon {
```

Comment:
I'm going to continue to call our this type of code as dangerous for the design of our system. If we need to know about specific button classes here in our icon component code for cosmetic reasons, that represents additional complexity and coupling. Going further, if we style an icon differently when it is used within a button using this approach we'll need to continue to consider each concrete usage of icon and account for them here, further expanding the coupling between other parts of our system and hindering our ability to make changes in the future.

We need to design code that can be understood, refactored, and deployed independently in a component first manner. To make this explicit, we would introduce an `<o-view>` component that does nothing but apply margin/padding/border styles and for the specific _implementation_ of an icon within a button, we would apply right margin. If the styling happened to be something specific to the icon, for example the width, height, or color we would consider a new variant such as `.ods-icon--xxl`.

To overcome complexity and coupling, we need to willfully ignore the context when writing code and that starts at the technical design phase of our work!

Reply:
While I'm 100% interested in getting us away from contextual styling (especially because it's not possible without breaching the ShadowDOM or incredibly careful use of `:host-context`), this is currently the simplest way for us to indicate always-on rules until we re-architect a lot of other things & have implementation parity with WCs.

Re: the solutions, Harry Roberts has been pretty vocal about different options over the years. This article is old, but not a ton has changed: https://csswizardry.com/2015/06/contextual-styling-ui-components-nesting-and-implementation-detail/ The Walton article he links speaks specifically to WCs as well. Part of our consideration has been the need to support composing all of these UIs in pure HTML/CSS. The current approach has allowed us to provide some level of programmatic detection without JS, but I agree that it's not a `1.0`-ready solve. 

For my part, I am leaning toward the Brad Frost way of thinking:

> I get this question a lot: how do I create a component that looks/behaves one way in one context, but another way in a second context? And I usually respond by telling them those should be two different components. People don’t like hearing this because it involves making and maintaining more design patterns, but ultimately that’s what you need to do if you want to be able to create robust, resilient design systems.

Instead of providing utility classes, contextual styling, or positioning components, Design teams should be incredibly discrete about the inventory of components and their variants. 

e.g. `.ods-button--icon` instead of `.ods-button .ods-icon` with both of them utilizing `@include icon-base` (or something similar, we have a lot of options!)

Brad's answer above doesn't speak to Harry's distinction between "look" and "behave" - that is, it feels weird to call something a discrete component when the only adjustment is a margin. However, I think that as Design alignment grows, there will be less and less variation, especially around cosmetics.

To that end, our current solution allows for easy contextual overrides (`.page--snowflake .button`). This, I think, is something *users* could employ to great affect, but is something the system shouldn't need to do.